src/main/java/org/jabref/logic/importer/WebFetchers.java

```diff
@@ -122,8 +126,65 @@ private WebFetchers() {
     }
 
     /**
-     * @return sorted set containing entry based fetchers
+     * @return an Hashmap containing predicate from id type and their id based fetchers
      */
+    public static HashMap<Predicate<String>,IdBasedFetcher> getHashMapPredicateIdBasedFetchers(ImportFormatPreferences importFormatPreferences) {
```

Comment:
I think it is better if the `IdBasedFetcher` interface is extended by a method `boolean matchIdentifier(string identifier)`, which then is implemented by each fetcher. In this way, the fetcher itself is responsible to decide which identifier it is able to match.